### PR TITLE
Update GTK packages (gtk2, gtk3, gnome_icon_theme)

### PR DESCRIPTION
**Pull request by JL2210**
Removes unneeded dependency `gtk3` from `gnome_icon_theme` and switches the dependencies.
Should fix most issues for missing image box in place of image.

### CODE REVIEW

File: packages/gtk2.rb [hunk 1]
```diff
@@ -1,38 +1,33 @@
 require 'package'
 
 class Gtk2 < Package
-  description 'Gtk+ 2.24 graphical user interface library.'
+  description 'GTK+ is a multi-platform toolkit for creating graphical user interfaces.'
```
Comment (uberhacker) [inline packages/gtk2.rb]: Much better description. 👍

File: packages/gtk2.rb [hunk 2]
```diff
@@ -1,38 +1,33 @@
...
   depends_on 'pango'
-  depends_on 'cairo'
   depends_on 'gdk_pixbuf'
+  depends_on 'cups'
   depends_on 'six' => :build
 
   def self.build
-    system "./configure --prefix=#{CREW_PREFIX} --libdir=#{CREW_LIB_PREFIX}"
+    system "./configure",
+           "--with-gdktarget=x11",
```
Comment (uberhacker) [inline packages/gtk2.rb]: If we add this option shouldn't we also add some X11 dependency or does that get included on a fresh install?

File: packages/gtk2.rb [hunk 3]
```diff
@@ -1,38 +1,33 @@
...
+           "--prefix=#{CREW_PREFIX}",
+           "--libdir=#{CREW_LIB_PREFIX}"
     system "make"
   end
 
   def self.install
-    system "make", "DESTDIR=#{CREW_DEST_DIR}", "install" # the steps required to install the package
+    system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
```
Comment (uberhacker) [inline packages/gtk2.rb]: I'm surprised you didn't change the double quotes to single quotes. ;)

File: packages/gtk3.rb [hunk 4]
```diff
@@ -1,47 +1,50 @@
...
   def self.install
     system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
+  end
+
+  def self.postinstall
     # generate schemas
-    system "mkdir -p #{CREW_DEST_PREFIX}/share/glib-2.0/schemas"
-    system "glib-compile-schemas #{CREW_DEST_PREFIX}/share/glib-2.0/schemas"
+    system "glib-compile-schemas #{CREW_PREFIX}/share/glib-2.0/schemas"
+    system "update-mime-database #{CREW_PREFIX}/share/mime"
```
Comment (uberhacker) [inline packages/gtk3.rb]: Do applications know to pull data from this directory?

File: packages/gtk3.rb [hunk 5]
```diff
@@ -1,47 +1,50 @@
...
 
   def self.build
     system "./configure",
+           "--enable-cups",
+           "--disable-debug",
+           "--enable-x11-backend",
+           "--enable-introspection",
            "--prefix=#{CREW_PREFIX}",
-           "--libdir=#{CREW_LIB_PREFIX}",
+           "--enable-wayland-backend",
```
Comment (uberhacker) [inline packages/gtk3.rb]: No wayland dependency needed?

Comment (JL2210) [reply packages/gtk3.rb]: No. My guess is that its dynamically loaded.

File: packages/gtk3.rb (hunk 4 above)
Comment (JL2210) [reply packages/gtk3.rb]: Yes.

File: packages/gtk2.rb (hunk 3 above)
Comment (JL2210) [reply packages/gtk2.rb]: Sorry. Forgot.

File: packages/gtk2.rb (hunk 2 above)
Comment (JL2210) [reply packages/gtk2.rb]: I don't think it's needed.

Comment (JL2210) [reply packages/gtk2.rb]: I don't think it's needed.

**Comment by uberhacker** [comment]
You are losing credibility @JL2210.  Turns out the "unneeded" gtk3 dependency is needed after all in gnome_icon_theme:
```
checking for gtk-update-icon-cache... no
configure: error: Could not find gtk-update-icon-cache
gnome_icon_theme failed to build: `./configure --prefix=/usr/local --enable-icon-mapping` exited with 1
```
Please, please, please test these things on a fresh install before you submit a PR.  If you cannot do that, then do not suggest changes.  It wastes a lot of my time.

**Comment by JL2210** [comment]
I removed gtk3 and it worked just fine. It might be one of the dependencies
of gtk3. Maybe gtk3 should be a build dependency?

**Comment by JL2210** [comment]
And you know I CANNOT test things on a fresh install. If I did that every
time I tested something, I'd be able to do this for about two days before
my "unlimited" (screw AT&T) data ran out.

**Comment by uberhacker** [comment]
Don't let your limited bandwidth stop you from a fresh install.  Just tarball your `/usr/local` directory and restore from backup.  That would be faster than downloading everything again anyway:
```
$ tar czvf fresh-crew.tar.gz /usr/local
... do your thing here ...
$ rm -rf /usr/local
$ tar xvf fresh-crew.tar.gz
```
Just remember to create a new tarball every time master is updated.

**Pull request merged**